eng/native/configureplatform.cmake

```diff
@@ -445,7 +445,7 @@ endif(CLR_CMAKE_TARGET_UNIX)
 # check if host & target os/arch combination are valid
 if (CLR_CMAKE_TARGET_OS STREQUAL CLR_CMAKE_HOST_OS)
     if(NOT(CLR_CMAKE_TARGET_ARCH STREQUAL CLR_CMAKE_HOST_ARCH))
-        if(NOT((CLR_CMAKE_HOST_ARCH_AMD64 AND CLR_CMAKE_TARGET_ARCH_ARM64) OR (CLR_CMAKE_HOST_ARCH_I386 AND CLR_CMAKE_TARGET_ARCH_ARM) OR (CLR_CMAKE_HOST_ARCH_AMD64 AND CLR_CMAKE_TARGET_ARCH_ARM) OR (CLR_CMAKE_HOST_ARCH_AMD64 AND CLR_CMAKE_TARGET_ARCH_I386)))
+        if(NOT((CLR_CMAKE_HOST_ARCH_AMD64 AND CLR_CMAKE_TARGET_ARCH_ARM64) OR (CLR_CMAKE_HOST_ARCH_I386 AND CLR_CMAKE_TARGET_ARCH_ARM) OR (CLR_CMAKE_HOST_ARCH_AMD64 AND CLR_CMAKE_TARGET_ARCH_ARM) OR (CLR_CMAKE_HOST_ARCH_AMD64 AND CLR_CMAKE_TARGET_ARCH_I386) OR (CLR_CMAKE_HOST_ARCH_ARM64 AND CLR_CMAKE_TARGET_ARCH_AMD64)))
```

Comment:
IMO, this check has outlived its usefulness. Removing this check means that the unsupported combinations will fail to build (run into one of the #error). Which usually happens when folks porting to new platforms fix.

Reply:
I agree that it would be better to delete this.

Reply:
makes sense